libs/vkd3d/vkd3d_private.h

```diff
@@ -1409,9 +1409,35 @@ static inline void d3d12_desc_copy_embedded_resource_payload(uint8_t *dst, const
 {
     /* Add special unrolled paths for the usual suspects. */
     if (size == 64)
+    {
+#ifdef __SSE2__
```

Comment:
Not sure if `d3d12_desc_copy_embedded_resource_payload` really adds that much value as an abstraction at this point, should we maybe have something like this in a common header instead and call those directly?
```c
static inline memcpy_nontemporal_64b(void* dst, const void* src) {
#ifdef __SSE2__
    ...
#else
    memcpy(dst, src, 64);
#endif
}
```

Maybe derp an always_inline there as well.

The current code kind of relies too much on the compiler being smart about inlining for my taste, which *probably* happens to work well right now but is by no means guaranteed (I've seen GCC not inline more trivial functions than this for no apparent reason, and if that happens we'll end up getting useless branching for a constant parameter).

Reply:
Sure, I'll refactor this so it does not require inlining + opts to figure it out.

Reply:
Rewrote this part now.